### DISCUSSION TITLE
Disable Event Emitting on services

### DISCUSSION THREAD

**Discussion by arcanys-ivan** [feathersjs/feathers#3198] [Q&A]
Hi, is there a way to disable the event emitting for specific services? Needed for some services that should not be emitting a respective data. I know it can be handle in publish and filter out the connection but its just too many since you have to specify every event names.

**Comment by daffl** [comment]
You can stop publishing all events on a service [as shown here](https://feathersjs.com/api/channels.html#service-publish-event-fn) like this:

```ts
// Prevent all events in the `password-reset` service from being published
app.service('password-reset').publish(() => null)
```

You can also disable event emitting entirely (internally as well) by setting [context.event](https://feathersjs.com/api/hooks.html#context-event) to `null` in a hook

```ts
const disableEvents = (context: HookContext) => {
  context.event = null
}
```

**Reply by arcanys-ivan** [reply]
Oh thanks for this one, I missed that line there
